### PR TITLE
Operation Spaghonut: Commence Spaghetti 

### PR DESCRIPTION
**Pull request by AlvCyktor**
Adds Spaghetti Donut into the game.
Spagonut
FullSpagonut

Sprite by Drywall Thief#2792 on discord, this PR is also their request.

:cl:  
rscadd: Spaghetti Donut

/:cl:

### CODE REVIEW

**Comment by TheLinuxOS** [comment]
Whoever came up with this deserves to be shot

File: code/modules/food_and_drinks/food/snacks_pastry.dm [hunk 1]
```diff
@@ -119,6 +119,19 @@
 	tastes = list("jelly" = 3, "donut" = 1, "fizzy tutti frutti" = 1)
 	filling_color = "#803280"
 
+/obj/item/reagent_containers/food/snacks/donut/spaghetti
+	name = "Spagh-o-nut"
+	desc = "An unholy mixture of carbs. It's a donut made out of spaghetti."
+	icon_state = "donut_spaghetti"
+	bonus_reagents = list(/datum/reagent/consumable/nutriment = 1)
```
Comment (SomeguyManperson) [inline code/modules/food_and_drinks/food/snacks_pastry.dm]:
```suggestion
	bonus_reagents = list(/datum/reagent/consumable/nutriment/vitamin = 1)
```

File: code/modules/food_and_drinks/food/snacks_pastry.dm [hunk 2]
```diff
@@ -119,6 +119,19 @@
...
+	tastes = list("Spaghetti"= 3, "Carbs" = 2, "Bewilderment" = 1)
+
+/obj/item/reagent_containers/food/snacks/donut/spaghetti/jelly
+	name = "'Jelly' Spagh-o-nut"
+	desc = "A Spaghetti Donut stuffed with ketchup."
+	icon_state = "jdonut_spaghetti"
+	bonus_reagents = list(/datum/reagent/consumable/nutriment = 1, /datum/reagent/consumable/ketchup = 2)
```
Comment (SomeguyManperson) [inline code/modules/food_and_drinks/food/snacks_pastry.dm]:
```suggestion
	bonus_reagents = list(/datum/reagent/consumable/nutriment/vitamin = 1, /datum/reagent/consumable/ketchup = 2)
```

**Comment by AlvCyktor** [comment]
> Whoever came up with this deserves to be shot

I do.

**Review by nesisknesis — COMMENTED** [review]
i dont know if the amount of noise this donut has is worth the concept so im gonna have to get back to you later

**Review by nesisknesis — APPROVED** [review]

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**